manifests/filter/open_tracing.pp

```diff
@@ -0,0 +1,180 @@
+# == Resource:  repose::filter::open_tracing
+#
+# This is a resource for enabling emitting traces to an OpenTracing tracer (current supports Jaeger).
+# Only usable in >= 8.8.3.0
+#
+# === Parameters
+#
+# [*ensure*]
+# Bool. Ensure config file present/absent
+# Defaults to <tt>present</tt>
+#
+# [*filename*]
+# String. Config filename.
+# Defaults to <tt>open-tracing.cfg.xml</tt>
+#
+# [*service_name*]
+# String. Application name
+# Defaults to <tt>repose</tt>
+#
+# [*connection_http*]
```

Comment:
if connection_* are mutually exclusive, why not convert them to `connection_type` and then prefix endpoint, username, password, token, and port with `connection_`, negate `host`, and potentially simplifying validation

```
validate_re($connection_type, ['http','udp'], 'connection_type parameter must be either http or udp')
if $connection_type == 'http' {
  validate_re($connection_endpoint, 'http?://', 'Must provide valid http:// endpoint for connection_endpoint for connecttion_type http')
  validate_string($connection_username)
  validate_string($connection_password)
  validate_string($connection_token)
} elif $connection_type == 'udp' {
  validate_re($connection_endpoint, '', 'Must provide valid hostname endpoint for connection_endpoint for connecttion_type udp')
  validate_integer($connection_port)
}
```

this should also pull forward to Puppet 4+'s Strict Typing capabilities.